src/Framework/IHasFilePath.cs

```diff
@@ -0,0 +1,10 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.Build.Framework
+{
+    public interface IHasSourceFilePath
```

Comment:
I would also not oppose making this an `internal` interface . . . theoretically as it is now it could work for a third-party task factory, but there are very few of those!

Reply:
Agree. I made this interface internal